.github/workflows/create-azure-self-hosted-runners.yml

```diff
@@ -135,9 +141,10 @@ jobs:
       with:
         resourceGroupName: ${{ secrets.AZURE_RESOURCE_GROUP }}
         template: ./azure-self-hosted-runners/azure-arm-template.json
-        parameters: ./azure-self-hosted-runners/azure-arm-template-example-parameters.json githubActionsRunnerRegistrationUrl="${{ env.ACTIONS_RUNNER_REGISTRATION_URL }}" githubActionsRunnerToken="${{ env.ACTIONS_RUNNER_TOKEN }}" postDeploymentPsScriptUrl="${{ env.POST_DEPLOYMENT_SCRIPT_URL }}" virtualMachineName=${{ steps.generate-vm-name.outputs.vm_name }} virtualMachineSize=Standard_D8pls_v5 publicIpAddressName1=${{ steps.generate-vm-name.outputs.vm_name }}-ip adminUsername=${{ secrets.AZURE_VM_USERNAME }} adminPassword=${{ secrets.AZURE_VM_PASSWORD }}
+        parameters: ./azure-self-hosted-runners/azure-arm-template-example-parameters.json githubActionsRunnerRegistrationUrl="${{ env.ACTIONS_RUNNER_REGISTRATION_URL }}" githubActionsRunnerToken="${{ env.ACTIONS_RUNNER_TOKEN }}" postDeploymentPsScriptUrl="${{ env.POST_DEPLOYMENT_SCRIPT_URL }}" virtualMachineName=${{ steps.generate-vm-name.outputs.vm_name }} virtualMachineSize=Standard_D8pls_v5 publicIpAddressName1=${{ steps.generate-vm-name.outputs.vm_name }}-ip adminUsername=${{ secrets.AZURE_VM_USERNAME }} adminPassword=${{ secrets.AZURE_VM_PASSWORD }} stopService={{ env.DEALLOCATE_IMMEDIATELY }}
```

Comment:
This is where the `{{` needs to be replaced with `${{`. But as I said, I'd like to avoid that construct in favor of native PowerShell environment variable expansion here (and elsewhere in this part of the workflow definition, in a separate commit).

Reply:
> This is where the `{{` needs to be replaced with `${{`. 

Oops! Good catch. Will fix.

> But as I said, I'd like to avoid that construct in favor of native PowerShell environment variable expansion here (and elsewhere in this part of the workflow definition, in a separate commit).

Unfortunately, that won't work. The `DEALLOCATE_IMMEDIATELY` env variable is only available in the Actions Runner context. When the VM is created, it does the following:

1. Download the post-deployment script from https://raw.githubusercontent.com/git-for-windows/git-for-windows-automation/main/azure-self-hosted-runners/post-deployment-script.ps1
2. Execute it _from within the VM_ (which doesn't have env variables like `DEALLOCATE_IMMEDIATELY`)

So we need to explicitly pass these parameters with their actual values to the `azure/arm-deploy` action, which in turn [passes it](https://github.com/git-for-windows/git-for-windows-automation/blob/37df492868dcb7f01ff8682e03d66a4badbfd853/azure-self-hosted-runners/azure-arm-template.json#L233-L260) to the `Microsoft.CustomScriptExtension`, which in turn will execute the two steps above.

Reply:
Sorry, I think I misunderstood actually:

> In general, I think we should always use the native environment variable expansion in scripts, i.e. `$env:XYZ` in PowerShell, `$XYZ` in Bash.

That makes sense to me. Just replacing `${{ env.XYZ }}` _everywhere_ in the workflow. Will do!

Reply:
How does [this](https://github.com/git-for-windows/git-for-windows-automation/pull/11/commits/52bab03c5999115facfb5da8685b682b574f4be8) look?

Reply:
> How does [this](https://github.com/git-for-windows/git-for-windows-automation/pull/11/commits/52bab03c5999115facfb5da8685b682b574f4be8) look?

It looks good! I did have to change the `ACTIONS_RUNNER_REGISTRATION_URL` lines because they only set the environment variable for subsequent steps, not the current one, but we no longer use it in a subsequent step, but in the current one.